pkg/ebpf/c/tracee.bpf.c

```diff
@@ -8047,9 +8073,16 @@ CGROUP_SKB_HANDLE_FUNCTION(proto_tcp_dns)
 CGROUP_SKB_HANDLE_FUNCTION(proto_udp_dns)
 {
     // submit DNS base event if needed (full packet)
-    if (should_submit_net_event(neteventctx, SUB_NET_PACKET_DNS)) {
+    if (should_submit_net_event(neteventctx, SUB_NET_PACKET_DNS))
         cgroup_skb_submit_event(ctx, neteventctx, NET_PACKET_DNS, FULL);
-        cgroup_skb_capture(); // capture udp dns packets
+
+    if (should_submit_net_event(neteventctx, SUB_NET_PACKET_IP) ||
+        should_submit_net_event(neteventctx, SUB_NET_PACKET_UDP) ||
+        should_submit_net_event(neteventctx, SUB_NET_PACKET_DNS)) {
+
+        // full dns header for capture (payload doesn't make much sense)
+        neteventctx->md.header_size = ctx->len;
+        cgroup_skb_capture(); // capture dns-udp, udp or ip packets
```

Comment:
ditto.

```suggestion
        cgroup_skb_capture(); // capture ip, udp or dns-udp packets
```

Reply:
This is in reverse order because that is how it should be going backwards (in the logic of capturing).